src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if (key.equals(entry.key)) {
+                entry.value = value;
+                return;
+            }
+        }
+        addEntry(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if (key.equals(entry.key)) {
+                return entry.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        for (Node<K, V> entry = table[0]; entry != null; entry = entry.next) {
+            if (entry.key == null) {
+                entry.value = value;
+                return;
+            }
+        }
+        addEntry(0, null, value, 0);
+    }
+
+    private void addEntry(int hash, K key, V value, int bucketIndex) {
+        Node<K, V> newNode = new Node<>(hash, key, value, table[bucketIndex]);
+        table[bucketIndex] = newNode;
+        size++;
+        if (size > table.length * LOAD_FACTOR) {
+            resize();
```

Comment:
Probably we can make resizing in the beginning of the method

Reply:
Still exists.